test/lib.api.v1.settings-controller.test.js

```diff
@@ -4,28 +4,103 @@ var request = require('supertest');
 var settingsController = require('../lib/api/v1/settings-controller');
 var botMaster = require('../lib/fetching/bot-master');
 var Source = require('../models/source');
+var config = require('../config/secrets.js');
+
+function clone(a) {
+  return JSON.parse(JSON.stringify(a));
+}
 
 describe('Settings controller', function() {
+  var twitterSettings;
+  var testSettings;
+
   before(function(done) {
     botMaster.kill();
     botMaster.addListeners('source', Source.schema);
     botMaster.addListeners('fetching', settingsController);
-    
+
     Source.create({nickname: 'one', media: 'dummy', keywords: 'one'});
     Source.create({nickname: 'two', media: 'dummy', keywords: 'two'});
     Source.create({nickname: 'three', media: 'dummy', keywords: 'three'});
-    
-    var config = require('../config/secrets');
+
+    twitterSettings =  clone(config.get().twitter);
+    testSettings = {
+      consumer_key: 'testing',
+      consumer_secret: 'test',
+      access_token: 'api',
+      access_token_secret: 'error',
+      configured: false,
+      on: false
+    };
+
     config.updateFetching(false, function(err){
       done();
     });
   });
 
+  describe('GET /api/v1/settings/twitter', function(){
```

Comment:
There are many code style issues in this file. Please read through and tidy it up. I realize some of them may have been done by someone else.